### PR TITLE
Convert `pi` commands to rzshell

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Convert all `pi` commands to rzshell
- Remove some unreachable code

**Test plan**

- CI is green
- Try different `pi` commands

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/1590

### CODE REVIEW

File: librz/core/cmd_descs/cmd_print.yaml [hunk 1]
```diff
@@ -452,15 +452,86 @@ commands:
...
+      - name: pir
+        summary: Print instructions using recursive disassembly algorithm
+        cname: print_instr_recursive
+        type: RZ_CMD_DESC_TYPE_ARGV_STATE
+        modes:
+          - RZ_OUTPUT_MODE_STANDARD
+          - RZ_OUTPUT_MODE_JSON
+        args: []
+      - name: pir.
+        summary: Print instructions using recursive disassembly algorithm
```
Comment (ret2libc) [inline librz/core/cmd_descs/cmd_print.yaml]: there should be a different description here, otherwise they seem the same.

**Review by ret2libc — APPROVED** [review]

File: librz/core/cmd/cmd_print.c [hunk 2]
```diff
@@ -6504,3 +6300,167 @@ RZ_IPI RzCmdStatus rz_print_key_mosaic_handler(RzCore *core, int argc, const cha
...
+	if (N == 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_core_disasm_pdi(core, N, 0, 'e');
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_instr_function_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	if (state->mode == RZ_OUTPUT_MODE_JSON) {
+		rz_core_cmdf(core, "pdfj");
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]: I think there is already an API for this command?

File: librz/core/cmd/cmd_print.c [hunk 3]
```diff
@@ -3343,21 +3313,18 @@ dsmap {
 }
 #endif
 
-static void disasm_until_ret(RzCore *core, ut64 addr, char type_print, const char *arg) {
+static void disasm_until_ret(RzCore *core, ut64 addr, int limit, RzOutputMode mode) {
 	int p = 0;
 	const bool show_color = core->print->flags & RZ_PRINT_FLAGS_COLOR;
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: use config api for the color

File: librz/core/cmd/cmd_print.c [hunk 4]
```diff
@@ -3343,21 +3313,18 @@ dsmap {
...
-	int i, limit = 1024;
-	if (arg && *arg && arg[1]) {
-		limit = rz_num_math(core->num, arg + 1);
-	}
+	int i;
 	for (i = 0; i < limit; i++) {
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: define the i into the for loop

**Review by wargio — CHANGES_REQUESTED** [review]
Please replace the eprintf at line `3350` inside `cmd_print.c` with `RZ_LOG_ERROR`

**Review by ret2libc — APPROVED** [review]

File: librz/core/cmd/cmd_print.c [hunk 5]
```diff
@@ -3391,30 +3357,17 @@ static void disasm_until_ret(RzCore *core, ut64 addr, char type_print, const cha
 	return;
 }
 
-static void func_walk_blocks(RzCore *core, RzAnalysisFunction *f, char input, char type_print, bool fromHere) {
-	RzListIter *iter;
-	RzAnalysisBlock *b = NULL;
+static void func_walk_blocks(RzCore *core, RzAnalysisFunction *f, bool fromHere, RzCmdStateOutput *state) {
 	const char *orig_bb_middle = rz_config_get(core->config, "asm.bb.middle");
 	rz_config_set_i(core->config, "asm.bb.middle", false);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c:3362]: This is set but never restored :/

**Review by wargio — APPROVED** [review]

**Pull request merged**